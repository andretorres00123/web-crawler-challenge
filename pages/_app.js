import App from 'next/app';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { compose } from 'redux';
import withRedux from 'next-redux-wrapper';
import { appWithTranslation } from '../i18n';
import { makeStore } from '../statemanagement/store';
import theme from '../styles/theme';

class MyApp extends App {

  static async getInitialProps({ Component, ctx }) {
    const pageProps = Component.getInitialProps ? await Component.getInitialProps(ctx) : {};

    return { pageProps };
  }

  render() {
    const { Component, pageProps, store } = this.props;
    return (
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Component {...pageProps} />
        </ThemeProvider>
      </Provider>
    );
  }
}

export default compose(
  withRedux(makeStore),
  appWithTranslation,
)(MyApp);
