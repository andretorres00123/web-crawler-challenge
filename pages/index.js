import React from 'react';
import { Box } from 'rebass';
import { connect } from 'react-redux';
import HomeHeader from '../components/widgets/HomeHeader';
import NewsList from '../components/widgets/NewsList';
import Actions from '../statemanagement/actions';

class Home extends React.Component {

  static async getInitialProps() {
    return {
      namespacesRequired: ['common'],
    };
  }

  componentDidMount() {
    const { getAllNews } = this.props;
    getAllNews();
  }

  render() {
    return (
      <Box>
        <HomeHeader />
        <NewsList />
      </Box>
    );
  }
}

const { getAllNews } = Actions;

export { Home };

export default connect(null, {
  getAllNews,
})(Home);
