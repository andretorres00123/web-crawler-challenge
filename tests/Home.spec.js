import React from 'react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Home } from '../pages/index';
import theme from '../styles/theme';

const props = {
  getAllNews: () => { },
};

const initialState = {
  news: {
    list: [],
  },
};

const mockStore = configureStore();
let store;

describe('Home page', () => {

  beforeEach(() => {
    store = mockStore(initialState);
  });
  test('should match with the snapshot', () => {
    const tree = renderer.create(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Home {...props} />
      </ThemeProvider>
    </Provider>).toJSON()
    expect(tree).toMatchSnapshot();
  });
});
