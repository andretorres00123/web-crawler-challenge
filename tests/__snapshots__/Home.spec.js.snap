// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should match with the snapshot 1`] = `
<div
  className="css-vurnku"
>
  <div
    className="css-1hwv0r3"
  >
    <div
      className="css-vurnku"
    >
      <div
        className="sc-AxjAm kAbRxt css-vurnku"
      >
        home-title
      </div>
      <div
        className="css-1lx7oj"
      >
         home-title-author
      </div>
    </div>
  </div>
  <div
    className="css-1b2tw1j"
  >
    <div
      className="sc-AxgMl hQfuHX"
    >
      <input
        checked={true}
        name="filters"
        onChange={[Function]}
        type="radio"
        value="all"
      />
      home-filters-all-value
      <input
        checked={false}
        name="filters"
        onChange={[Function]}
        type="radio"
        value="longTitles"
      />
      home-filters-long-titles-value
      <input
        checked={false}
        name="filters"
        onChange={[Function]}
        type="radio"
        value="shortTitles"
      />
      home-filters-short-titles-value
    </div>
  </div>
  <div
    className="css-h1u7m2"
  >
    <div
      className="css-hphxtc"
    >
      <div
        className="css-qpwnd"
      >
        <div
          className="sc-AxjAm kAbRxt css-1dnv0i3"
        >
          news-column-name-rank
        </div>
      </div>
      <div
        className="css-6ha5j8"
      >
        <div
          className="sc-AxjAm kAbRxt css-1dnv0i3"
        >
          news-column-name-title
        </div>
      </div>
      <div
        className="css-e7q5pv"
      >
        <div
          className="sc-AxjAm kAbRxt css-1dnv0i3"
        >
          news-column-name-comments
        </div>
      </div>
      <div
        className="css-e7q5pv"
      >
        <div
          className="sc-AxjAm kAbRxt css-1dnv0i3"
        >
          news-column-name-points
        </div>
      </div>
    </div>
  </div>
</div>
`;
